core/services/gateway/handlers/functions/allowlist.go

```diff
@@ -188,6 +191,7 @@ func (a *onchainAllowlist) updateFromContractV1(ctx context.Context, blockNum *b
 	if err != nil {
 		return errors.Wrap(err, "unexpected error during functions_allow_list.NewTermsOfServiceAllowList")
 	}
+	a.lggr.Debugw("allowlist route ID", tosID, "allowlist contract address", tosAddress)
```

Comment:
This is not how "Debugw" works. It needs a message first and then a list of pairs (label, value). https://pkg.go.dev/go.uber.org/zap

Let's also log tosID before erroring out from GetContractById() as it might be the reason for failure there.
Similarly, log tosAddress before erroring out from NewTermsOfServiceAllowList.